hw/src/ledger.rs

```diff
@@ -335,6 +344,53 @@ impl Manager {
 	}
 }
 
+pub struct EventHandler {
+	ledger: Weak<Manager>,
+}
+
+impl EventHandler {
+	pub fn new(ledger: Weak<Manager>) -> Self {
+		Self { ledger: ledger }
+	}
+
+	fn is_valid_product(&self, device: &libusb::Device) -> Result<(), Error> {
+		let desc = device.device_descriptor()?;
+		let vendor_id = desc.vendor_id();
+		let product_id = desc.product_id();
+
+		if vendor_id == LEDGER_VID && LEDGER_PIDS.contains(&product_id) {
+			Ok(())
+		} else {
+			Err(Error::InvalidDevice)
+		}
+	}
+}
+
+
+impl libusb::Hotplug for EventHandler {
+	fn device_arrived(&mut self, device: libusb::Device) {
+		println!("ledger arrived");
+		Duration::from_millis(1000);
```

Comment:
this line does nothing, stray println above

Reply:
Oops, should be ```thread::sleep(Duration::from_millis(1000))``` ofc